src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -2388,4 +2390,43 @@ private function addHtmlSanitizerSection(ArrayNodeDefinition $rootNode, callable
             ->end()
         ;
     }
+
+    private function addImportMapSection(ArrayNodeDefinition $rootNode, callable $enableIfStandalone): void
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('importmap')
+                    ->info('Import Map configuration')
+                    ->{$enableIfStandalone('symfony/import-map', ImportMapManager::class)}()
+                    ->fixXmlConfig('import_map')
+                    ->children()
+                        ->scalarNode('path')
+                            ->info('The path of the importmap.php file.')
+                            ->defaultValue('%kernel.project_dir%/importmap.php')
+                        ->end()
+                        ->scalarNode('polyfill')
+                            ->info('URL of the ES Module Polyfill to use, false to disable.')
+                            ->defaultNull()
+                        ->end()
+                        ->scalarNode('vendor_dir')
+                            ->info('The directory where to store JavaScript vendors.')
+                            ->defaultValue('%kernel.project_dir%/vendor/javascript/')
+                        ->end()
+                        ->scalarNode('public_dir')
+                            ->info('The directory where to store the publicly exposed local JavaScript modules.')
+                            ->defaultValue('%kernel.project_dir%/public/javascript/')
+                        ->end()
+                        ->scalarNode('base_url')
+                            ->info('The base URL to access JavaScript modules.')
+                            ->defaultValue('/javascript/')
```

Comment:
I couldn't get the paths setup correctly without a hack - am I doing something wrong?

A) I'm importing the `importmap.xml` route - so the route path is `/assets/{path}`
B) `importmap.php`, I have an entry like:

```
'app' => [
    'path' => 'assets/app.js',
],
```

Without any config, the importmap renders on the page as `"app": "/javascript/app.1234.js"` - notice that it doesn't align with the `/assets/{path}` from the route. If I correct that with `framework.importmap.base_url: '/assets/'`, it hits the controller, but the path isn't found as `ImportMapManager::getPathForUrl()` is passed simply `app.1234.js`, but the key in `$this->urlToPath` is `assets/app.1234.js`.

The `assets/` and `javascripts/` not lining up by default, I think, is a bug - I like `assets/` better. But what about the last part about `getPathForUrl()` - is that a bug or am I missing some setup?

Reply:
It was indeed a bug in the controller. It is now fixed.
I prefer `javascript/` because it looks more modern, only ESMs are currently supported, and it aligns with what Rails does. But I can revert to `assets/` if necessary.